src/components/ui/Filters.jsx

```diff
@@ -0,0 +1,111 @@
+import {
+    Box, Button, Drawer, Stack, TextField, Typography
+} from "@mui/material";
+import React, {useState} from "react";
+import {FilterListOff, FilterList, Close} from "@mui/icons-material";
+import DeleteIcon from "@mui/icons-material/Delete";
+
+/**
+ *
+ * @param filterOptions
+ * @param filters {{field: value}}
+ * @param onChangeFilters {(field:string, value) => void}
+ * @param onClearFilters {()=>void}
+ * @returns {Element}
+ * @constructor
+ */
+export const Filters = ({filterOptions, filters, onChangeFilters, onClearFilters}) => {
+    const [isOpen, setIsOpen] = useState(false);
+    const drawerWidth = "20%";
+
+    const handleClearFilters = () => {
+        setIsOpen(false);
+        onClearFilters();
+    }
+    const RenderFilterOptions = (filter) => {
+        const {type, field, label, options} = filter
+        const clearButton = <Button aria-label="delete" color="warning"
+                                    onClick={() => onChangeFilters(field, undefined)} startIcon={<DeleteIcon/>}>
+            Clear
+        </Button>
+
+        switch (type) {
+            case "buttons":
+                return (<Stack key={filter.label} direction="column" spacing={1}>
+                    {label && <Typography variant="caption" component="div" sx={{
+                        textTransform: "capitalize", justifyContent: "space-between"
+                    }}>
+                        {label}
+                    </Typography>}
+                    <Box sx={{display: "flex", flexWrap: "wrap", gap: 1}}>
+                        {options.map(option => {
+                            return (<Button
+                                key={crypto.randomUUID()}
```

Comment:
Instead of that random uuid here you could make use of the index from the map function I guess we could have same behaviour